server/src/test/java/org/elasticsearch/snapshots/SnapshotResiliencyTests.java

```diff
@@ -235,6 +240,16 @@ public void createServices() {
     @After
     public void verifyReposThenStopServices() {
         try {
+            clearDisruptionsAndAwaitSync();
+
+            final StepListener<CleanupRepositoryResponse> cleanupResponse = new StepListener<>();
+            client().admin().cluster().cleanupRepository(
```

Comment:
I wonder if we should check before the clean-up whether the repo is not corrupted (i.e. does not have meta files pointing at non-existing data files).

Reply:
> i.e. does not have meta files pointing at non-existing data files

This is not something the cleanup will resolve ever. The root `index-N` points at the uuid named generations of shard-level metadata now. Neither of these is changed in content (only a new index-N with the same content as the previous file is written) as a result of the cleanup action.

-> I think it's fine to test things this way around so long as we don't do any repairs in the cleanup. We do the same in all other tests as well (all the `SharedClusterSnapshotRestoreIT` and such).

Reply:
The point is to have stronger assertions, asserting something about the state of the repository before we do any clean-up (and also something that will hold at any point during snapshotting / snapshot deletion etc., which we could check at any time). Having two separate sets of assertions also better shows what's being guaranteed in a situation without clean-up, and what the clean-up is essentially adding. You can do this in a follow-up, I feel however that mixing the two into one here weakens the coverage of these tests.

Reply:
Makes sense, will add that in a follow up :)